conda/api.py

```diff
@@ -1,5 +1,110 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-from .core.solve import Solver  # NOQA
-Solver = Solver  # lgtm [py/redundant-assignment]
+from .common.constants import NULL
+from .core.package_cache_data import PackageCacheData as _PackageCacheData
+from .core.prefix_data import PrefixData as _PrefixData
+from .core.solve import Solver as _Solver, DepsModifier as _DepsModifier
+from .core.subdir_data import SubdirData as _SubdirData
+from .models.channel import Channel
+
+
+DepsModifier = _DepsModifier
+
+
+class Solver(object):
+
+    def __init__(self, prefix, channels, subdirs=(), specs_to_add=(), specs_to_remove=()):
+        self._internal = _Solver(prefix, channels, subdirs, specs_to_add, specs_to_remove)
+
+    def solve_final_state(self, deps_modifier=NULL, prune=NULL, ignore_pinned=NULL,
+                          force_remove=NULL):
+        return self._internal.solve_final_state(deps_modifier, prune, ignore_pinned,
+                                                force_remove)
+
+    def solve_for_diff(self, deps_modifier=NULL, prune=NULL, ignore_pinned=NULL,
+                       force_remove=NULL, force_reinstall=False):
+        return self._internal.solve_for_diff(deps_modifier, prune, ignore_pinned,
+                                             force_remove, force_reinstall)
+
+    def solve_for_transaction(self, deps_modifier=NULL, prune=NULL, ignore_pinned=NULL,
+                              force_remove=NULL, force_reinstall=False):
+        return self._internal.solve_for_transaction(deps_modifier, prune, ignore_pinned,
+                                                    force_remove, force_reinstall)
+
+
+class SubdirData(object):
+
```

Comment:
We need a docstring here, that also explains why this `Solver` wrapper is needed, or intended for?